src/airgap/harness-airgap-images.sh

```diff
@@ -54,38 +78,64 @@ fi
 
 process_tgz_file() {
   local file=$1
+  local load_result
   echo "Processing $file..."
-  local image_info
-  local load_output
+  debug_log "Processing $file...This can take a while"
+  
+  #Ensures there is enough storage space for docker load
+  local required_space=$(du -ks "$file" | awk '{print $1}')
+  local available_space=$(df -Pk . | awk 'NR==2 {print $4}')
+  if [ "$required_space" -gt "$available_space" ]; then
+    debug_log "Insufficient disk space to load $file"
+    failed_images+=("$file")
+    ((fail_count++))
+    return 1
+  fi
+
+  debug_log "Loading Docker image from $file..."
+  load_result=$(docker load -q -i "$file" 2>&1)
+  
+  if [[ "$load_result" =~ *"Error processing tar file"* ]]; then
+    echo "Error loading image from $file: $load_result"
+    failed_images+=("$file")
+    ((fail_count++))
+    error_occurred=true
+    return 1
+  fi
 
-  # Load the Docker image from a .tgz file
-  load_output=$(docker load -i "$file" 2>&1)
+  # Check the exit status of docker load
+  local exit_status=$?
```

Comment:
This seems to be duplicated from the above check

Reply:
Yes, the above check matches text for corrupted file, but this exist_status is more generic and will also catch corrupted file scenario. I'll remove previous check.